components/camel-aws/camel-aws2-ddb/src/main/docs/aws2-ddb-component.adoc

```diff
@@ -266,8 +266,29 @@ URI:
 
 [source,java]
 ----------------------------------------------------
-from("direct:start")
-.to("aws2-ddb://domainName?amazonDDBClient=#client");
+public class MyRouteBuilder extends RouteBuilder {
+
+    private String accessKey = "myaccessKey";
+    private String secretKey = "secretKey";
+
+    @Override
+    public void setCamelContext(CamelContext camelContext) {
```

Comment:
It is not good practice to do code in a setter - keep setter as plain getter/setter.

Instead you can create a method with this logic, and call this method from configure()

Reply:
Thanks for the advice. 
I changed it to register the aws client in the configure()

` getCamelContext().getRegistry().bind("client", client);`